### PR TITLE
8300054: Use static_assert in basic_types_init

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change to use `static_assert` when applicable inside a function.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300054](https://bugs.openjdk.org/browse/JDK-8300054): Use static_assert in basic_types_init

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11971/head:pull/11971` \
`$ git checkout pull/11971`

Update a local copy of the PR: \
`$ git checkout pull/11971` \
`$ git pull https://git.openjdk.org/jdk pull/11971/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11971`

View PR using the GUI difftool: \
`$ git pr show -t 11971`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11971.diff">https://git.openjdk.org/jdk/pull/11971.diff</a>

</details>

### CODE REVIEW

**Review by stefank — APPROVED** [review]
Looks good. One nit:

File: src/hotspot/share/utilities/globalDefinitions.cpp [hunk 1]
```diff
@@ -155,11 +155,11 @@ void basic_types_init() {
...
   // These are assumed, e.g., when filling HeapWords with juints.
-  assert(is_power_of_2(sizeof(juint)), "juint must be power of 2");
-  assert(is_power_of_2(HeapWordSize), "HeapWordSize must be power of 2");
-  assert((size_t)HeapWordSize >= sizeof(juint),
+  static_assert(is_power_of_2(sizeof(juint)), "juint must be power of 2");
+  static_assert(is_power_of_2(HeapWordSize), "HeapWordSize must be power of 2");
+  static_assert((size_t)HeapWordSize >= sizeof(juint),
          "HeapWord should be at least as large as juint");
-  assert(sizeof(NULL) == sizeof(char*), "NULL must be same size as pointer");
+  static_assert(sizeof(NULL) == sizeof(char*), "NULL must be same size as pointer");
```
Comment (stefank) [inline src/hotspot/share/utilities/globalDefinitions.cpp:162]: Maybe fix alignment here.

**Review by stefank — APPROVED** [review]

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**